tree/ntuple/inc/ROOT/RNTupleReader.hxx

```diff
@@ -82,6 +83,8 @@ private:
    /// Retrieving descriptor data from an RNTupleReader is supposed to be for testing and information purposes,
    /// not on a hot code path.
    std::optional<ROOT::RNTupleDescriptor> fCachedDescriptor;
+   /// We know that the RNTupleReader is always reading a single RNTuple, so the number of entries is fixed.
+   mutable std::atomic<ROOT::NTupleSize_t> fCachedNEntries = kInvalidNTupleIndex;
```

Comment:
Why does this need to be atomic? It only gets written under a lock and it's then only read afterwards, no?

Reply:
See above, changed.